api/services/freshtrakApiService.ts

```diff
@@ -0,0 +1,27 @@
+import request from 'request-promise';
+import getLogger from '../utils/logger';
+import { FRESHTRAK_API_HOST } from '../utils/constants';
+
+const log = getLogger('freshtrakApiService');
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function getFreshTrakEvents(agencyId: any): Promise<any>{
```

Comment:
Do we know what the response object looks like?
```suggestion
async function getFreshTrakEvents(agencyId: number): Promise<FreshtrakAgenciesResponseInterface>{
```

Reply:
We shouldn't be ignoring that eslint rule...

Reply:
Just tested without it and can be removed.  SCOSservice.makeSCOSRequest has the same ignore statement and no interface on what is returned.
'''
// eslint-disable-next-line @typescript-eslint/no-explicit-any
async function makeSCOSRequest(body: any): Promise<any> {
'''

Reply:
The API JSON response for both the freshtrak and SCOS APIs is large and will change.  Typing the response could cause problems when that occurs.  We are mocking the response in our tests but only for data we are using and that data is being typed.

Reply:
Proposed solution: The freshtrakResponseDto interface will work here and is what the caller is expecting.  This interface only includes what the application requires to avoid breaking changes.  Null is returned when a server error is detected and logged.

```
async function getFreshTrakEvents(agencyId: any): Promise<freshtrakResponseDto | null> {
```

Reply:
> Just tested without it and can be removed. SCOSservice.makeSCOSRequest has the same ignore statement and no interface on what is returned.
> 
> // eslint-disable-next-line @typescript-eslint/no-explicit-any
> async function makeSCOSRequest(body: any): Promise<any> {

I don't remember this being added - but kind of defeats the purpose of using TS. 



> Proposed solution: The freshtrakResponseDto interface will work here and is what the caller is expecting. This interface only includes what the application requires to avoid breaking changes. Null is returned when a server error is detected and logged.
> 
> ```
> async function getFreshTrakEvents(agencyId: number): Promise<freshtrakResponseDto | null> {
> ```

This looks great, thanks for looking into it. 
Also `null` is not returned on error, it is actually `undefined` since nothing is returned in the catch block so would look like `Promise<freshtrakResponseDto | void>`.

Reply:
Locally I had changed the function to explicitly return null.  Seems more clear to me than leaving it hanging.

Reply:
Yea I was looking at that - returning the error might be useful too